rtengine/dcraw.cc

```diff
@@ -10007,10 +10009,23 @@ static inline uint32_t DNG_FP24ToFloat(const uint8_t * input) {
 static void expandFloats(Bytef * dst, int tileWidth, int bytesps) {
   if (bytesps == 2) {
     uint16_t * dst16 = (uint16_t *) dst;
+#ifndef __F16C__
     uint32_t * dst32 = (uint32_t *) dst;
     for (int index = tileWidth - 1; index >= 0; --index) {
-      dst32[index] = DNG_HalfToFloat(dst16[index]);
+        dst32[index] = DNG_HalfToFloat(dst16[index]);
+    }
+#else
+    float * dst32 = (float *) dst;
+    int index = tileWidth - 4;
+    for (; index >= 0; index -= 4) {
+        __m128i halfFloatv = _mm_loadu_si128((__m128i_u*)&dst16[index]);
+        STVFU(dst32[index], _mm_cvtph_ps(halfFloatv));
```

Comment:
Let's assume the load is expensive and the conversion is cheap. Then there's room for improvement, as you load 8 half-floats, but you convert and store only 4, always reloading 4 unused ones. Maybe walk in 8 float steps and use something like `_mm_srli_si128` on `halfFloatv`.

Reply:
good idea :+1: